code/datums/gamemodes/infestation.dm

```diff
@@ -33,6 +33,21 @@
 	for(var/i in GLOB.xeno_jelly_pod_turfs)
 		new /obj/structure/xeno/resin_jelly_pod(i, XENO_HIVE_NORMAL)
 
+	scale_weapons()
+
+/datum/game_mode/infestation/proc/scale_weapons(pop_override)
+	if(!isnum(pop_override))
+		pop_override = length(GLOB.alive_human_list)
```

Comment:
this proc is run once on post setup... so doesn't include the majority of players that latejoin.